mayastor/src/bdev/nexus/nexus_child_error_store.rs

```diff
@@ -105,61 +113,138 @@ impl NexusErrStore {
         self.next_record_index =
              // consider using a mask if we restrict the size to a power-of-2
             (self.next_record_index + 1) % self.records.len();
-        // debug!("added record - buffer is {}", &self);
     }
 
-    pub fn query(
+    #[inline]
+    fn is_wanted_record(
+        record: &NexusChildErrorRecord,
+        io_op_flags: u32,
+        io_error_flags: u32,
+        target_timestamp: Option<Instant>,
+    ) -> bool {
+        let found_op = match record.io_op {
+            io_type::READ => (io_op_flags & NexusErrStore::READ_FLAG) != 0,
```

Comment:
this can be refined using match guards: https://doc.rust-lang.org/reference/expressions/match-expr.html#match-guards

Reply:
Refactored using match guards.